fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java

```diff
@@ -0,0 +1,144 @@
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class WorkGroupClassifier {
+    public static final Pattern UseRolePattern = Pattern.compile("^\\w+$");
+    public static final Set<String> QUERY_TYPES = Arrays.stream(QueryType.values()).map(Enum::name).collect(Collectors.toSet());
+    private long id;
+    private String user;
+    private String role;
+    private Set<QueryType> queryTypes;
+    private SubnetUtils.SubnetInfo sourceIp;
+    private long workgroupId;
+
+    public long getWorkgroupId() {
+        return workgroupId;
+    }
+
+    public void setWorkgroupId(long workgroupId) {
+        this.workgroupId = workgroupId;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public String getUser() {
+        return user;
+    }
+
+    public void setUser(String user) {
+        this.user = user;
+    }
+
+    public String getRole() {
+        return role;
+    }
+
+    public void setRole(String role) {
+        this.role = role;
+    }
+
+    public Set<QueryType> getQueryTypes() {
+        return queryTypes;
+    }
+
+    public void setQueryTypes(Set<QueryType> queryTypes) {
+        this.queryTypes = queryTypes;
+    }
+
+    public SubnetUtils.SubnetInfo getSourceIp() {
+        return sourceIp;
+    }
+
+    public void setSourceIp(SubnetUtils.SubnetInfo sourceIp) {
+        this.sourceIp = sourceIp;
+    }
+
+    public TWorkGroupClassifier toThrift() {
+        TWorkGroupClassifier classifier = new TWorkGroupClassifier();
+        classifier.setId(id);
+        classifier.setWorkgroup_id(workgroupId);
+        if (user != null) {
+            classifier.setUser(user);
+        }
+        if (role != null) {
+            classifier.setRole(role);
+        }
+        if (queryTypes != null) {
+            classifier.setQuery_types(queryTypes.stream().map(QueryType::name).collect(Collectors.toSet()));
+        }
+        if (sourceIp != null) {
+            classifier.setSource_ip(sourceIp.getCidrSignature());
+        }
+        return classifier;
+    }
+
+    public boolean isSatisfied(String user, String role, QueryType queryType, String sourceIp) {
+        if (!isVisible(user, role, sourceIp)) {
+            return false;
+        }
+        if (this.queryTypes != null && !this.queryTypes.isEmpty() && !this.queryTypes.contains(queryType)) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isVisible(String user, String role, String sourceIp) {
+        if (this.user != null && !this.user.equals(user)) {
+            return false;
+        }
+        if (this.role != null && !this.role.equals(role)) {
+            return false;
+        }
+        if (!this.sourceIp.isInRange(sourceIp)) {
+            return false;
+        }
+        return true;
+    }
+
+    public double weight() {
```

Comment:
This weight algorithm seems strange.

Reply:
the narrower the classifer is, the more heavy it weights